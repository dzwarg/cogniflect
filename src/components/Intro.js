import React from 'react';
import {Button, Col, Container, FormControl, InputGroup, Jumbotron, Row} from 'react-bootstrap';
import {LinkContainer} from 'react-router-bootstrap';
import '../styles/Intro.css';

const Intro = ({appState, handleChange, handleClick}) => {
  const userTeamCode = appState.get('userTeamCode');
  
  return (
    <Container>
      <Row>
        <Col>
          <Jumbotron>
            <h1>This is the Cogniflector</h1>
            <p>
              This application will ask you a short series of questions to you and your team.
            </p>
          </Jumbotron>
        </Col>
      </Row>
      <Row>
        <Col xs={12}>
          <p>
            This application will take you through the steps to evaluate your own cognitive reflection. The process is simple:
          </p>
          <ol>
            <li>Answer a series of multiple choice questions individually.</li>
            <li>Answer a series of multiple choice questions in groups of 3 or more.</li>
            <li>The scores from step <b>1</b> and <b>2</b> are compared.</li>
          </ol>
        </Col>
      </Row>
      <Row>
        <Col xs={{span:6,offset:3}} md={{span:4,offset:4}} lg={{span:2,offset:5}}>
          <p>
            <InputGroup>
              <InputGroup.Prepend>
                <InputGroup.Text>#</InputGroup.Text>
              </InputGroup.Prepend>
              <FormControl type="text" placeholder="team code" onChange={handleChange} value={userTeamCode} />
            </InputGroup>
          </p>
        </Col>
      </Row>
      <Row>
        <Col xs={{span:6,offset:3}} md={{span:4,offset:4}} lg={{span:2,offset:5}}>
          <p className="centered">
            <LinkContainer to="/question/1">
              <Button onClick={handleClick(userTeamCode)} variant="success">Let&apos;s get started!</Button>
            </LinkContainer>
          </p>
        </Col>
      </Row>
    </Container>
  );
};

export default Intro
