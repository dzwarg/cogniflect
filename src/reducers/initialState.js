import {fromJS} from 'immutable';

const initialState = fromJS({
  teamCode: '',
  userTeamCode: '',
  assessmentType: 'individual',
  questionCursor: 0,
  questions: [
    {
      id: 0,
      text: "In a lake, there is a patch of lily pads. Every day, the patch doubles in size. If it takes 48 days for the patch to cover the entire lake, how long would it take for the patch to cover half of the lake?",
      truth: 1,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '24' },
        { id: 1, text: '47' },
        { id: 2, text: '96' },
        { id: 3, text: '49' }
      ],
      next: 1
    },
    {
      id: 1,
      text: "A bat and a ball cost $1.10 in total. The bat costs $1.00 more than the ball. How much does the ball cost?",
      truth: 0,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '$0.05' },
        { id: 1, text: '$0.06' },
        { id: 2, text: '$0.10' },
        { id: 3, text: '$0.50' }
      ],
      next: 2
    },
    {
      id: 2,
      text: "If it takes 5 machines 5 minutes to make 5 widgets, how long would it take 100 machines to make 100 widgets?",
      truth: 0,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '5 minutes' },
        { id: 1, text: '100 minutes' },
        { id: 2, text: '500 minutes' },
        { id: 3, text: '10000 minutes' }
      ],
      next: 3
    },
    {
      id: 3,
      text: "If you’re running a race and you pass the person in second place, what place are you in?",
      truth: 1,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '1st place' },
        { id: 1, text: '2nd place' },
        { id: 2, text: '3rd place' },
        { id: 3, text: 'Last place' }
      ],
      next: 4
    },
    {
      id: 4,
      text: "A farmer had 15 sheep and all but 8 died. How many are left?",
      truth: 2,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '6' },
        { id: 1, text: '7' },
        { id: 2, text: '8' },
        { id: 3, text: '9' }
      ],
      next: 5
    },
    {
      id: 5,
      text: "Emily’s father has three daughters. The first two are named April and May. What is the third daughter’s name?",
      truth: 3,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: 'June' },
        { id: 1, text: 'August' },
        { id: 2, text: 'Summer' },
        { id: 3, text: 'Emily' }
      ],
      next: 6
    },
    {
      id: 6,
      text: "How many cubic feet of dirt are there in a hole that is 3’ deep x 3’ wide x 3’ long?",
      truth: 0,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '0 ft^3' },
        { id: 1, text: '3 ft^3' },
        { id: 2, text: '9 ft^3' },
        { id: 3, text: '27 ft^3' }
      ],
      next: null
    }
  ],
  team: {
    members: 0,
    synced: 0
  }
});

export default initialState;