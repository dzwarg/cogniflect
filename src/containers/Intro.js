import { connect } from 'react-redux'
import actions from '../actions'
import Intro from '../components/Intro'

const mapStateToProps = (state) => ({
  appState: state.appState
})

const mapDispatchToProps = (dispatch) => ({
  handleChange: (evt) => {
    dispatch(actions.set(['userTeamCode'], evt.target.value));
  },
  handleClick: (teamCode) => (evt) => {
    dispatch(actions.set(['teamCode'], teamCode));
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(Intro)
export {mapStateToProps, mapDispatchToProps}
