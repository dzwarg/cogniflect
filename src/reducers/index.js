import { combineReducers } from 'redux'
import initialState from './initialState'; 

const appState = (state = initialState, action) => {
  switch (action.type) {
    case 'SET':
      return state.setIn(action.payload.path, action.payload.value);
    default:
      return state
  }
}

export default combineReducers({
  appState
});