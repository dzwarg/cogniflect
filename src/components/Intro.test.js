import React from 'react';
import Intro from './Intro';
import {Button, FormControl} from 'react-bootstrap';
import reducers from '../reducers';
import {Map} from 'immutable';

const initialState = {appState: Map({userTeamCode: 'foo', teamCode: 'foo'})};

describe('dumb component', () => {
  it('without crashing', () => {
    const component = Intro({...initialState, handleClick:()=>{}})
    expect(component).toMatchSnapshot();
  });
});
