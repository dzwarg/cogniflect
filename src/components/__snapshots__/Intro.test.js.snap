// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Intro matches snapshot 1`] = `
<ForwardRef(Bootstrap(Container))>
  <ForwardRef(Bootstrap(Row))>
    <ForwardRef(Bootstrap(Col))>
      <ForwardRef(Bootstrap(Jumbotron))>
        <h1>
          This is the Cogniflector
        </h1>
        <p>
          This application will ask you a short series of questions to you and your team.
        </p>
      </ForwardRef(Bootstrap(Jumbotron))>
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(Row))>
  <ForwardRef(Bootstrap(Row))>
    <ForwardRef(Bootstrap(Col))
      xs={12}
    >
      <p>
        This application will take you through the steps to evaluate your own cognitive reflection. The process is simple:
      </p>
      <ol>
        <li>
          Answer a series of multiple choice questions individually.
        </li>
        <li>
          Answer a series of multiple choice questions in groups of 3 or more.
        </li>
        <li>
          The scores from step 
          <b>
            1
          </b>
           and 
          <b>
            2
          </b>
           are compared.
        </li>
      </ol>
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(Row))>
  <ForwardRef(Bootstrap(Row))>
    <ForwardRef(Bootstrap(Col))
      lg={
        Object {
          "offset": 5,
          "span": 2,
        }
      }
      md={
        Object {
          "offset": 4,
          "span": 4,
        }
      }
      xs={
        Object {
          "offset": 3,
          "span": 6,
        }
      }
    >
      <p>
        <ForwardRef(Bootstrap(InputGroup))>
          <ForwardRef(Bootstrap(InputGroupPrepend))>
            <ForwardRef(Bootstrap(InputGroupText))>
              #
            </ForwardRef(Bootstrap(InputGroupText))>
          </ForwardRef(Bootstrap(InputGroupPrepend))>
          <ForwardRef(ContextTransform)
            placeholder="team code"
            type="text"
            value="foo"
          />
        </ForwardRef(Bootstrap(InputGroup))>
      </p>
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(Row))>
  <ForwardRef(Bootstrap(Row))>
    <ForwardRef(Bootstrap(Col))
      lg={
        Object {
          "offset": 5,
          "span": 2,
        }
      }
      md={
        Object {
          "offset": 4,
          "span": 4,
        }
      }
      xs={
        Object {
          "offset": 3,
          "span": 6,
        }
      }
    >
      <p
        className="centered"
      >
        <LinkContainer
          activeClassName="active"
          exact={false}
          replace={false}
          strict={false}
          to="/question/1"
        >
          <ForwardRef(Bootstrap(Button))
            variant="success"
          >
            Let's get started!
          </ForwardRef(Bootstrap(Button))>
        </LinkContainer>
      </p>
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(Row))>
</ForwardRef(Bootstrap(Container))>
`;
